app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -33,59 +34,24 @@
+ * It contains 2 children fragments based on the state of the page
+ * @see HomeSelectStateFragment
+ * @see HomeBaseStateFragment
+ * @author Owen Cooke, Jared Drueco, Lukas Bonkowski, Sami Jagirdar
  */
-public class HomeFragment extends Fragment implements FilterCallback {
-    private CollectionReference itemRef;
-    private ListView itemListView;
-    private ArrayList<Item> itemList = new ArrayList<>();
-    private Set<Filter> appliedFilters = new HashSet<>();
-    private ArrayAdapter<Item> itemAdapter;
-
-    /**
-     * This constructs a new HomeFragment with the appropriate list of items
-     * @param itemRef A reference to the firestore collection containing the items to display
-     */
-    public HomeFragment(CollectionReference itemRef) {
-        this.itemRef = itemRef;
-        itemList = new ArrayList<>();
-    }
-
-    /**
-     * @param inflater           The LayoutInflater object that can be used to inflate
-     *                           any views in the fragment,
-     * @param container          If non-null, this is the parent view that the fragment's
-     *                           UI should be attached to.  The fragment should not add the view itself,
-     *                           but this can be used to generate the LayoutParams of the view.
-     * @param savedInstanceState If non-null, this fragment is being re-constructed
-     *                           from a previous saved state as given here.
-     * @return the home fragment view containing the inventory list
-     */
-    @Override
-    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        // Inflate the fragment's layout
-        View rootView = inflater.inflate(R.layout.fragment_home, container, false);
-
-        itemListView = rootView.findViewById(R.id.item_list);
-        itemAdapter = new ItemAdapter(getContext(), itemList);
-        itemListView.setAdapter(itemAdapter);
-
-        itemRef.addSnapshotListener(this::setupItemListener);
-
-        View filterButton = rootView.findViewById(R.id.filter_dropdown_button);
-        filterButton.setOnClickListener(this::showFilterMenu);
-
-        return rootView;
-
-    }
+public abstract class HomeFragment extends Fragment {
+    protected CollectionReference itemRef;
+    protected ListView itemListView;
+    protected ArrayList<Item> itemList;
+    protected ArrayAdapter<Item> itemAdapter;
 
     /**
      * This method updates the itemAdapter with changes in the firestore database and creates new
      * item objects
      * @param querySnapshots The updated information on the inventory from the database
      * @param error Non-null if an error occurred in Firestore
      */
-    private void setupItemListener(QuerySnapshot querySnapshots, FirebaseFirestoreException error) {
+    protected void setupItemListener(QuerySnapshot querySnapshots, FirebaseFirestoreException error) {
```

Comment:
This should be able to go into the HomeBaseState

Reply:
Yeah you're right, I had it in Home because initially, both HomeBase and HomeSelect were using the listener. Will now move to HomeBase.